### ISSUE TITLE
Site selection cannot handle sites with the same name

### ISSUE THREAD

**Issue by davidye007** [OutdoorRD/trails-viz#260]
In the `West Cascades` project, there are two site names (Bear Lake, Quartz Creek) that polygons with different site ids share.
`Quartz Creek` is associated with site id `38` and `68`.
`Bear Lake` is associated with site id `19` and `180`.

The problem is the user cannot select the polygon `Quartz Creek` with site id `38`. If the user does select that polygon, the dashboard assumes `Quartz Creek` with site id `68` is selected.

Similarly, the user cannot select the polygon 'Bear Lake' with side id `19`. If it's selected, the dashboard assumes `Bear Lake` with site id `180` is selected.

This happens because the `projectSites` dictionary inside `MapDiv.vue` uses trail names as the key. When there exist multiple sites (site ids) with the same trail name, the earlier site added to `projectSites` will be overwritten by the sites added later. Effectively, `Quartz Creek` with site id `38` and `Bear Lake` with side id `19` does not exist in `projectSites`. Thus, these sites can never be selected. 

A plausible fix is appending duplicate trail names with the site id since site ids are unique for each site. An example would be `Bear Lake` vs. `Bear Lake (180)`, or `Quartz Creek` vs. `Quartz Creek (68)`. This way, even sites that have the same trail name will still be distinct and exist separately inside `projectSites`. This allows users to select either one of the `Quartz Creek` sites (same for Bear Lake).

**Comment by EmiliaH** [comment]
Thanks for pointing this out @davidye007! This sounds like it may the same cause of this [issue Quartz Creek Trail Portal](https://github.com/orgs/OutdoorRD/projects/31?pane=issue&itemId=79167628&issue=OutdoorRD%7Ctrails-viz-data%7C53) (which I just added to the roadmap). 

This issue will likely come up in other Forests and is likely to be even more problematic if we have multiple Forests/management lands in one project with trail-shed polygons/sites.

**Comment by davidye007** [comment]
@EmiliaH Exactly! The Quartz Creek transportation/teleportation you noted in https://github.com/OutdoorRD/trails-viz-data/issues/53 is caused by this issue. Since this issue also impacts visualizing chatbot locations, at the moment I've been appending the site-id to sites that have duplicate trail names.

**Comment by EmiliaH** [comment]
Alright, that seems like a reasonable fix for now! It may be worth considering other options but let's talk about this tomorrow. @sgwinder, you may also have some thoughts on solutions

**Comment by sgwinder** [comment]
Thanks for digging into this, David! 

> This happens because the `projectSites` dictionary inside `MapDiv.vue` uses trail names as the key. 

Do you know why this is? (eg is there some reason that we need the key to be trail names, in order to show up correctly on the dashboard or something?)

My initial response is that we should be using `siteid` as the key, rather than trail name. And that then we shouldn't have this problem.

**Comment by davidye007** [comment]
@sgwinder The keys of `projectSites` are currently used for the `Search Trail` functionality (e.g. dropdown list of trail names) and for polygon selection. Both these functionalities should still work if we used `siteid` as `projectSites` keys instead.

I think using `siteid` as the key rather than trail name is a good approach, as `siteid` is unique for each site. This way, all our site polygons will work, regardless of duplicate trail names. I can move forward with this approach.

Something this issue also brings up is we should probably find a way to distinguish sites with the same trail name to users. For example, if we have multiple sites with the trail name `Bear Lake`, when the user searches for `Bear Lake`, two identical `Bear Lake` options should show up in the dropdown (same with Quartz Creek). I think there should be a quick way for the user to identify which `Bear Lake` is which. @EmiliaH mentioned there will likely be even more duplicate trail names when multiple projects are combined/brought together. Though I think this is more of an issue of distinguishing duplicate trail names for users, than fixing site selection itself.

**Comment by sgwinder** [comment]
Thanks David. If we use `siteid` as the key, will users still be able to see `Trail name` in the `Search Trail` functionality? This seems important to me.

And yes, I agree it would be nice to distinguish duplicate names - but it's tricky to know how to do this generally. I like your suggestion above about appending the siteids (eg `Bear Lake (180)`). That would make the dashboard more usable for us (internal ORD users), and would at least make it so that we could describe the different polygons to external users (even if they don't have context to understand the siteids themselves).

**Comment by davidye007** [comment]
@sgwinder Yes, if we use `siteid` as the key, users will still be able to see `Trail name` in the `Search Trail` functionality. Using `siteid` as keys resolves the issue of duplicate sites not being added to `projectSites`. 

Though this by itself did not resolve the 'site teleportation' or trail search functionality. While all sites with duplicate trail names technically exist on the dashboard, and the trail search functionality lists two trails with the same name, the user still isn't able to select the second duplicate trail name site. This is because much of the search and site selection functionality are implemented using `strings`. As a result, the first trail in `projectSites` that matches the given `string` (e.g. `'Bear Lake') `will be selected. Thus, whether the user selects the first or second 'Bear Lake' site, the search function only knows the user selected `'Bear Lake'` and will always return the first match in `projectSites` (hence still no access the second 'Bear Lake' site).

To resolve issues with polygon site selection, I updated the site selection function to handle `site objects`, rather than `strings`. This fixes the issue with 'site teleportation'.

To resolve issues with `Search Trail` functionality, I made sure each trail name is unique. This is accomplished by appending each duplicate trail name with their site id. We need unique trail names because the search functionality is `string` based. To match the searched `string` to a particular `site object`, me must ensure each `every` corresponds to a single `site objects`. And a simple way to do this is ensuring unique trail names.

These two solutions resolve the issue with no being able to select polygons/search sites with duplicate trail names.
